user-cli/rust/src/main.rs

```diff
@@ -0,0 +1,140 @@
+use data_encoding::HEXLOWER;
+use ring::{digest, pbkdf2};
+use std::num::NonZeroU32;
+
+use nix::unistd::Uid;
+
+mod backend;
+use crate::backend::device::*;
+use crate::backend::dm::*;
+
+#[macro_use]
+extern crate clap;
+use clap::App;
```

Comment:
Reorder and regroup the use and module declarations in the following form.

```rust
// Import external APIs first
use data_encoding::HEXLOWER;
use ring::{digest, pbkdf2};
use std::num::NonZeroU32;
use nix::unistd::Uid;
use clap::App;

// Then, declare the modules of this crate
mod backend;

// Then, import internal APIs
use crate::backend::device::*;
use crate::backend::dm::*;
```

By the way, remember to run `cargo fmt`, which will sort use declarations automatically.

Reply:
Thanks! Module declarations are reordered.